sdk/core/azure-core/inc/azure/core/http/curl/curl_session.hpp

```diff
@@ -319,14 +319,25 @@ namespace Azure { namespace Core { namespace Http {
       return eof && m_sessionState != SessionState::PERFORM;
     }
 
+    /**
+     * @brief All connection will be open and requested to maintain the channel open to re-use the
+     * connection.
+     *
+     * @remark This option can be disabled from the transport adapter options. When disabled, the
+     * session won't return connections to the connection pool. Connection will be clean as soon as
+     * the request is completed.
+     *
+     */
+    bool m_keppAlive = true;
+
   public:
     /**
      * @brief Construct a new Curl Session object. Init internal libcurl handler.
      *
      * @param request reference to an HTTP Request.
      */
-    CurlSession(Request& request, std::unique_ptr<CurlNetworkConnection> connection)
-        : m_connection(std::move(connection)), m_request(request)
+    CurlSession(Request& request, std::unique_ptr<CurlNetworkConnection> connection, bool keppAlive)
```

Comment:
Is this ctor intended to be public (as in accessible to the end user)? What happens if the Curl options and  this bool disagree?

Reply:
No.
Maybe I can move all of this APIs to ::Details .. since nothing of this is public.
The only public stuff is the transportAdapter implementation. The session, connection and connectionPool are private and used only by the transportAdapter